### ISSUE TITLE
byteswap.c has a lot of unused code and confusion about GNU vs. non-GNU

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-bacio#68]
In byteswap.c (https://github.com/NOAA-EMC/NCEPLIBS-bacio/blob/develop/src/byteswap.c) there are the following functions:

```
static int
simple_swap_64(void *data,size_t len)
static int
simple_swap_32(void *data,size_t len)
static int
simple_swap_16(void *data,size_t len)

```
And also:

```
static int
macro_swap_64(void *data,size_t len)
static int
macro_swap_32(void *data,size_t len)
static int
macro_swap_16(void *data,size_t len)

```

The code documentation says that the macro_* versions are the same, but use the "GNU macros". These are GNU extension macros bswap_16, bswap_32, bswap_64.

The documentation says that the macro_* versions of the functions will be used on GNU systems. But in fact, a mix is used, and there is no difference between systems:

```
int
fast_byteswap(void *data,int bytes,size_t count)
{
  switch(bytes) {
  case 1: return 1;
  case 2: return simple_swap_16(data,count);
  case 4: return simple_swap_32(data,count);
  case 8: return macro_swap_64(data,count);
  default: return 0;
  }
}
```

Are the GNU macros available on Intel compilers? Apparently, because we are using one for 64 bit data. But not for 16 or 32.

@junwang-noaa you are listed as one of the authors of this code. Do you remember what is going on here?

My intention is to delete the unused code and keep what is used.

**Comment by junwang-noaa** [comment]
It looks to me this piece of code was added later, I am not sure why we need these macro_swaps.

**Comment by edwardhartnett** [comment]
This was resolved in #66. I will close this issue.
